lib/cache/cache.go

```diff
@@ -772,7 +772,7 @@ func (c *Cache) fetchAndWatch(ctx context.Context, retry utils.Retry, timer *tim
 	// by receiving init event first.
 	select {
 	case <-watcher.Done():
-		return trace.ConnectionProblem(watcher.Error(), "watcher is closed")
+		return trace.ConnectionProblem(watcher.Error(), "watcher is closed: %v", watcher.Error())
```

Comment:
Kind of a weird API that `trace.ConnectionProblem()` requires you to provide the underlying error twice.

Reply:
I agree. This is actually a problem with any `trace` error that takes an error and a message

https://github.com/gravitational/trace/blob/master/errors.go#L383
https://github.com/gravitational/trace/blob/master/errors.go#L618

We should probably address this in `trace` at some point - but doing so would cause duplicates in logs since this is likely not the only place in the code doing this to get around the current limitation